nifi-commons/nifi-web-client-api/src/main/java/org/apache/nifi/web/client/api/StandardMultipartFormDataStreamBuilder.java

```diff
@@ -0,0 +1,183 @@
+     * @return Builder
+     */
+    @Override
+    public MultipartFormDataStreamBuilder addPart(final String name, final HttpContentType httpContentType, final InputStream inputStream) {
+        Objects.requireNonNull(name, "Name required");
+        Objects.requireNonNull(httpContentType, "Content Type required");
+        Objects.requireNonNull(inputStream, "Input Stream required");
+
+        final Matcher nameMatcher = ALLOWED_NAME_PATTERN.matcher(name);
+        if (nameMatcher.matches()) {
+            final Part part = new Part(name, httpContentType, inputStream);
+            parts.add(part);
+        } else {
+            throw new IllegalArgumentException("Name contains characters outside of ASCII character set");
+        }
+
+        return this;
+    }
+
+    /**
+     * Add Part with field name and byte array source
+     *
+     * @param name Name field of part to be added
+     * @param httpContentType Content-Type of part to be added
+     * @param bytes Byte array content of part to be added
+     * @return Builder
+     */
+    @Override
+    public MultipartFormDataStreamBuilder addPart(final String name, final HttpContentType httpContentType, final byte[] bytes) {
+        Objects.requireNonNull(bytes, "Byte Array required");
+        final InputStream inputStream = new ByteArrayInputStream(bytes);
+        return addPart(name, httpContentType, inputStream);
+    }
+
+    private InputStream getPartInputStream(final Part part, final String footer) {
+        final String partHeaders = getPartHeaders(part);
+        final InputStream headersInputStream = new ByteArrayInputStream(partHeaders.getBytes(HEADERS_CHARACTER_SET));
+        final InputStream footerInputStream = new ByteArrayInputStream(footer.getBytes(HEADERS_CHARACTER_SET));
+        final Enumeration<InputStream> inputStreams = Collections.enumeration(List.of(headersInputStream, part.inputStream, footerInputStream));
+        return new SequenceInputStream(inputStreams);
+    }
+
+    private String getPartHeaders(final Part part) {
+        final StringBuilder headersBuilder = new StringBuilder();
+
+        final String contentDispositionHeader = CONTENT_DISPOSITION_HEADER.formatted(part.name);
+        headersBuilder.append(contentDispositionHeader);
+
+        final String contentType = part.httpContentType.getContentType();
+        final String contentTypeHeader = CONTENT_TYPE_HEADER.formatted(contentType);
+        headersBuilder.append(contentTypeHeader);
+
+        headersBuilder.append(CARRIAGE_RETURN_LINE_FEED);
+        return headersBuilder.toString();
+    }
+
+    private record MultipartHttpContentType(String contentType) implements HttpContentType {
+        @Override
+        public String getContentType() {
+            return contentType;
+        }
+    }
+
+    private record Part(
+            String name,
+            HttpContentType httpContentType,
+            InputStream inputStream
+    ) {
+
+    }
```

Comment:
More nit picky but is a whole new line necessary if there is no implementation
```suggestion
    ) {
    }
```